src/renderers/webgl-fallback/WebGLBackend.js

```diff
@@ -36,6 +36,23 @@ class WebGLBackend extends Backend {
 
 		const glContext = ( parameters.context !== undefined ) ? parameters.context : renderer.domElement.getContext( 'webgl2' );
 
+		function onContextLost( event ) {
+
+			event.preventDefault();
+
+			const contextLossInfo = {
+				api: 'WebGL',
+				message: event.statusMessage || 'Unknown reason',
+				reason: null,
+				originalEvent: event
+			};
+
+			renderer.onDeviceLost( contextLossInfo );
+
+		}
+
+		renderer.domElement.addEventListener( 'webglcontextlost', onContextLost, false );
```

Comment:
It would be good to remove this event listener when `renderer.dispose()` is called.

Do you mind implementing a `dispose()` method for the WebGL backend that is called from `Renderer.dispose()`?

Reply:
We don't have a `dispose()` method in the backends.
I could add an abstract one in the `Backend` class and on both backends, also I'm not sure if adding an `onContextLost` event outside the `WebGLBackend` Class, just like we do in the `WebGLRenderer`, will not break the tree-shaking?

For the moment I just remove the listener directly into the onContextLost (if that's possible?):
https://github.com/mrdoob/three.js/pull/29767/commits/e97021c548642b6850bf2f11c6ccc6c182430155#diff-2dc5b01c4242a7be2dc61d5019c5e6e60b231903addc2b221d495d480bffca48R43-R53

Reply:
Not sure this is ideal. Normally you want to keep the listener added since after a context restore the context might be lost again. The event listener should only be removed when the renderer isn't used anymore (so when calling `dispose()`.

> We don't have a dispose() method in the backends. I could add an abstract one in the Backend class and on both backends,

That sounds good to me. I'm not aware how the event listeners should break tree shaking if we use `removeEventListener()` in `dispose()`.